integration/nsmtesting/tools.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package nsmtesting provides k8s specific testing asserts
+package nsmtesting
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/edwarnicke/exechelper"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+)
+
+// NSMTesting is k8s specific asset tool
+type NSMTesting struct {
+	t      *testing.T
+	client *kubernetes.Clientset
+}
+
+// New creates new NSMTesting
+func New(t *testing.T) *NSMTesting {
+	return &NSMTesting{
+		t: t,
+	}
+}
+
+// K8s returns k8s client
+func (t *NSMTesting) K8s() *kubernetes.Clientset {
+	return t.client
+}
+
+// Exec executes command
+func (t *NSMTesting) Exec(cmd string) {
```

Comment:
Why are we creating indirection here to another object when its simple enough to simply do this inline in the test...
out,err := exechelper.Output(cmd,exechelper.WithStdout(logrus.Writer())
require.NoError(t,err)
...

Fun story about exechelper and WithStdout and WithStderr ... they [multiplex](https://github.com/edwarnicke/exechelper/blob/master/options.go#L81), so you can add more Writers at will... and since [exechelper.Output[(https://github.com/edwarnicke/exechelper/blob/master/exechelper.go#L92) is just adding a buffer using WithStdout... you can vastly simplify :)

Reply:
> Why are we creating indirection here to another object when its simple enough to simply do this inline in the test...
out,err := exechelper.Output(cmd,exechelper.WithStdout(logrus.Writer())
require.NoError(t,err)

To reduce lines count. For example, your suggestion has two lines to check output when it can be done with some helper like NSMTesting in one line :)

Reply:
> Fun story about exechelper and WithStdout and WithStderr ... they multiplex, so you can add more Writers at will... and since [exechelper.Output[(https://github.com/edwarnicke/exechelper/blob/master/exechelper.go#L92) is just adding a buffer using WithStdout... you can vastly simplify :)

It is a good point, will fix.

Reply:
> > Why are we creating indirection here to another object when its simple enough to simply do this inline in the test...
> > out,err := exechelper.Output(cmd,exechelper.WithStdout(logrus.Writer())
> > require.NoError(t,err)
> 
> To reduce lines count. For example, your suggestion has two lines to check output when it can be done with some helper like NSMTesting in one line :)
> 
> In short, my point is tests should be maxing shortest.

Two lines -> one line isn't really worth the indirection and non-locality :)  There's totally a place where it becomes worth it... but not IMHO that low :)

Reply:
The discussion has moved to https://github.com/networkservicemesh/integration-k8s-kind/pull/2#discussion_r459820551